spec/ics-003-connection-semantics/README.md

```diff
@@ -318,24 +330,27 @@ function connOpenTry(
     abortTransactionUnless(validateConnectionIdentifier(desiredIdentifier))
     abortTransactionUnless(consensusHeight < getCurrentHeight())
     expectedConsensusState = getConsensusState(consensusHeight)
-    expected = ConnectionEnd{INIT, desiredIdentifier, getCommitmentPrefix(), counterpartyClientIdentifier,
+    abortTransationUnless(
+      provedIdentifier === desiredIdentifier ||
```

Comment:
I'm running into issues implementing this as well. 

chainA sets the connection pointing at chainB with `proviedIdentifier`

chainB is verifying the connection set by chainA and setting its own connection pointing at chainB if everything is valid. This connection will use `desiredIdentifier` so why do `desiredIdentifier` and `provedIdentifier` need to be equal? This breaks our existing implementation:

chainA sets the connection on its chain with "connectionAID"

chainB wants to set the connection on its chain with "connectionBID" 

that would fail this if state. Was our implementation wrong? Am I missing something?

Reply:
`provedIdentifier` is the counterparty identifier chain A selected for chain B, if chain A chose a specific identifier chain B needs to use it or the handshake will fail.

chain A can use whatever identifier it wants on its own side for its own connection